sktime/clustering/partitioning/_lloyds_partitioning.py

```diff
@@ -0,0 +1,247 @@
+        self.init_algorithm = init_algorithm
+        self.random_state = random_state
+
+        self._centers: NumpyArray = None
+        self._init_algorithm = None
+        self._metric = None
+        self._random_state = None
+
+    def _fit(self, X: NumpyArray) -> None:
+        """
+        Method that is used to fit the time series k
+        partition model on dataset X
+
+        Parameters
+        ----------
+        X: Numpy_Array
+            Numpy array of series to train the model on
+
+        Returns
+        -------
+        self
+            Fitted estimator
+        """
+        self._random_state = check_random_state(self.random_state)
+        center_algo: BaseClusterCenterInitializer = self._init_algorithm(
+            X, self.n_clusters, self.calculate_new_centers, self._random_state
+        )
+        self._centers = center_algo.initialize_centers()
+
+        for _ in range(self.max_iter):
+            self.__update_centers(X)
+
+    def _predict(self, X: NumpyArray) -> NumpyArray:
+        """
+        Method used to perform a prediction from the trained
+        model
+
+        Parameters
+        ----------
+        X: Numpy_Array
+            Numpy_Array containing the time series to
+            predict clusters for
+
+        Returns
+        -------
+        Numpy_Array
+            Index of the cluster each sample belongs to
+        """
+        return self.__cluster_data(X)
+
+    def get_centers(self) -> NumpyArray:
+        """
+        Method used to get the centers of the clustering
+        algorithm
+
+        Returns
+        -------
+        Numpy_Array
+            Containing the values of the centers
+        """
+        return self._centers
+
+    def calculate_new_centers(self, cluster_values: NumpyArray) -> NumpyArray:
+        """
+        Method to be implemented by parent defining how centers
+        are calculated based on each iteration of k_partition
+
+        Parameters
+        ----------
+        cluster_values: Numpy_Array
+            Values to derive a center from (values in a cluster)
+
+        Returns
+        -------
+        Numpy_Array
+            Single value that is determined to be the center of
+            the series
+        """
+        raise NotImplementedError("abstract method")
+
+    def _check_params(self, X: NumpyArray):
+        """
+        Method used to check the parameters passed
+
+        Parameters
+        ----------
+        X: Numpy_Array
+            Dataset to be validate parameters against
+        """
+        if isinstance(self.init_algorithm, str):
+            self._init_algorithm = TimeSeriesLloydsPartitioning._init_algorithms[
+                self.init_algorithm
+            ]
+
+        if isinstance(self.metric, str):
+            self._metric = TimeSeriesLloydsPartitioning._metric_dict[self.metric]
+
+        super(TimeSeriesLloydsPartitioning, self)._check_params(X)
+
+    def __cluster_data(self, X: NumpyArray) -> List[List[int]]:
```

Comment:
```suggestion
    def _cluster_data(self, X: NumpyArray) -> List[List[int]]:
```

Reply:
This must have got readded by mistake when i was fixing whatever happened with the pr the other day good spot